cpp/oneapi/dal/table/backend/interop/host_homogen_table_adapter.cpp

```diff
@@ -334,14 +249,9 @@ bool host_homogen_table_adapter<Data>::check_column_index_in_range(const block_i
 template <typename Data>
 host_homogen_table_adapter<Data>::host_homogen_table_adapter(const homogen_table& table,
                                                              status_t& stat)
-        // The following const_cast is safe only when this class is used for read-only
-        // operations. Use on write leads to undefined behaviour.
-        : base(daal::data_management::DictionaryIface::equal,
-               ptr_data_t{ const_cast<Data*>(table.get_data<Data>()), daal_object_owner(table) },
-               table.get_column_count(),
-               table.get_row_count(),
-               stat),
-          is_rowmajor_(table.get_data_layout() == data_layout::row_major) {
+        : NumericTable(table.get_column_count(), table.get_row_count()) {
```

Comment:
Implicit conversion from signed to unsigned: `std::int64_t` -> `std::size_t`. Consider use of `detail::integral_cast`, defined in `dal/detail/common.hpp`

Reply:
fixed